lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +30,115 @@ const _changesRequiredOutput = """
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  Map<String, bool> inTransition = {};
+  bool foundReactOrOverReact = false;
+
+  final react16CodemodLogger = Logger('over_react_codemod.react16');
+  react16CodemodLogger.onRecord.listen((LogRecord record) {
+    print(record.message);
+  });
+  react16CodemodLogger
+      .info('Checking if project needs to run $react16CodemodName...');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e) {
+    if (e is FileSystemException) {
+      react16CodemodLogger
+          .warning('Could not find pubspec.yaml, exiting codemod.');
+    } else if (e is YamlException || e is ArgumentError) {
+      react16CodemodLogger
+          .warning('pubspec.yaml is unable to be parsed, exiting codemod.');
+    }
+    exitCode = 0;
+  }
+  for (var package in packagesToCheckFor.entries) {
+    var constraint = getDependencyVersion(package.key);
+    if (constraint != null) {
+      foundReactOrOverReact = true;
+      react16CodemodLogger
+          .info('Found ${package.key} with version ${constraint}');
+      // Found it so lets add it to the in transition list to false until its
+      // validated that we know it is in transtition.
+      inTransition[package.key] = false;
+    }
+    if (constraint != null &&
+        !constraint.isAny &&
+        constraint.allowsAny(package.value[0]) &&
+        constraint.allowsAny(package.value[1])) {
+      inTransition[package.key] = true;
+    }
+  }
+  if (foundReactOrOverReact) {
+    react16CodemodLogger.info(inTransition);
+    if (inTransition.isNotEmpty &&
+        !inTransition.values.any((val) => val == false)) {
+      react16CodemodLogger.info('Starting $react16CodemodName...');
+      final query = FileQuery.dir(
+        pathFilter: isDartFile,
+        recursive: true,
+      );
+      exitCode = runInteractiveCodemodSequence(
+        query,
+        [
+          ReactDomRenderMigrator(),
+          ReactStyleMapsUpdater(),
+        ],
+        args: args,
+        defaultYes: true,
+        changesRequiredOutput: _changesRequiredOutput,
+      );
+
+      final logger = Logger('over_react_codemod.fixmes');
+      for (var dartFile in query.generateFilePaths()) {
+        final dartSource = File(dartFile).readAsStringSync();
+        if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
+          logger.severe(
+              'over_react_codemod validation comments are unaddressed in $dartFile');
+          exitCode = 1;
+        }
+      }
+      react16CodemodLogger.info('Finished Running $react16CodemodName!');
+    } else {
+      react16CodemodLogger.warning(
+          'pubspec.yaml does not have transition versions of react or over_react, exiting codemod.');
```

Comment:
#nit semicolon reads better here
```suggestion
          'pubspec.yaml does not have transition versions of react or over_react; exiting codemod.');
```

Reply:
Also, we should probably include a message here informing the user to run our pubspec upgrader codemod (from https://github.com/Workiva/over_react_codemod/pull/39) if they want to proceed with this codemod.